utils/s2n_safety.h

```diff
@@ -95,13 +95,20 @@ static inline void* trace_memcpy_check(void *restrict to, const void *restrict f
 #define GUARD_GOTO( x , label ) do {if ( (x) < 0 ) goto label;} while (0)
 #define GUARD_PTR( x )          do {if ( (x) < 0 ) return NULL;} while (0)
 
+/* Similar to GUARD, but preserves the blocking error code. -2 is left to preserve backwards compatibility .*/
+#define GUARD_NONBLOCKING( x )                              \
+  do {                                                      \
+    int __tmp_r = (x);                                      \
+    if (__tmp_r == -2 || __tmp_r == S2N_ERR_BLOCKED) {      \
```

Comment:
can we make `-2` a constant?

Reply:
Added a callback return code, S2N_CALLBACK_BLOCKED which is -2.